hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.controller.js

```diff
@@ -307,12 +311,13 @@ define([
     }
 
     /**
-     * Converts a date string into a Date object (if string is not empty)
+     * Converts a date string into a Date object.
+     * If the date is not valid it returns the same date string provided.
      *
      * @todo This should probably happen inside the service that returns the data #
```

Comment:
Please add another `@todo` here, the function should not return `{Date|String}` but since it is like it was before, we lack test coverage and this PR is already big, lets postpone it till the next wave of refactoring.